common/cpp/km_u16.cpp

```diff
@@ -387,3 +388,39 @@ double u16tof(KMX_WCHAR* str16) {
   std::string str = string_from_u16string(str16);
   return strtof(str.c_str(), &pEnd);
 }
+
+/**
+ * @brief Trim whitespace from the start (left) of a string
+ * @param p Pointer to u16string
+ * @return Pointer to the first non-whitespace character in the string
+ */
+const KMX_WCHAR* u16ltrim(const KMX_WCHAR* p) {
+  while(p && iswspace(*p)) p++;
+  return p;
+}
```

Comment:
This is simple and doesn't modify the input, but I wonder if, for consistency with `u16rtrim` and `u16trim`, we should actually memmove the non-whitespace to start of `p`? Then we can say that the original pointer reference is always to the start of the memory block, which can be helpful for memory management.

This would also avoid the de-consting we see in u16trim

Reply:
Good idea ... the symmetry is better. Done.